modules/user_accounts/php/edit_user.class.inc

```diff
@@ -272,14 +272,40 @@ class Edit_User extends \NDB_Form
         // multi-site UPDATE
         $uid           = $user->getId();
         $us_curr_sites = $values['CenterIDs'];
+        $userNewCenterIDs = array_map(
+            function ($val) {
+                return new \CenterID($val);
+            },
+            $values['CenterIDs']
+        );
         if (!$this->isCreatingNewUser()) {
+            // Prevent deletion of rules/sites editor may not have access to edit.
+            if (!$editor->hasPermission('user_accounts_multisite')) {
+                // Collect CenterID(s) of user being modified.
+                $userCenterIDs = $user->getCenterIDs();
+                // Collect CenterID(s) of the editor modifying sites.
+                $editorCenterIDs = $editor->getCenterIDs();
+                // If the editor does not have access to all of the
+                // user's sites the difference between the two arrays
+                // will be the sites we additionally need to add to
+                // the user being modified.
+                $differenceOfCenterIDs = array_diff(
+                    $userCenterIDs,
```

Comment:
`$userCenterIDs` should only contain values belonging to the editor. It should be `array_intersect($userNewCenterIDs, $editorCenterIDs);`

Reply:
@xlecours are you suggesting that we should modify the user class' `getCenterIDs()` function to only return the sites that the editor has access to? (that's what im understanding here)

Reply:
No.

I made my comment at the wrong place, mentioning the wrong variable... my mistake.

What I meant was:
Because `$userNewCenterIDs` is basically the values received in the POST request, it could be any values if a malicious user forge the request. So, to prevent that user from adding sites not belonging to the editor, we should make sure that the final set of sites to add (`$userNewCenterIDs`) is composed of A) the sites the editor can't modify that already belong to the editee (that is `$differenceOfCenterIDs`), and B) the sites that were POSTed that must belong to the editor (that is the modification I am asking).

Something like:
```
                $userNewCenterIDs      = array_merge(
                    array_intersect($userNewCenterIDs, $editorCenterIDs)
                    $differenceOfCenterIDs
                );
``` 

Sorry for the confusion.

Reply:
thanks @xlecours